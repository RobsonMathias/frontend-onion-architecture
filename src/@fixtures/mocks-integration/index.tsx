export * from './store.mock';
export * from './adapters.mock';
export * from './../mocks/http.mock';
export * from './../mocks/router.mock';
