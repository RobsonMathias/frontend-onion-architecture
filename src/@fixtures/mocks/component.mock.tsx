import React from 'react';

export function Mock() {
  return <div />;
}

export const mockComponent = (componentsName: string[] = []) => {
  let composed: { [key: string]: React.ReactNode } = {};
  componentsName.forEach((name: string) => {
    composed[name] = Mock;
  });
  return composed;
};
