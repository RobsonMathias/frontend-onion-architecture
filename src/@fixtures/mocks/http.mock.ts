export const mockHttp = {
  patch: jest.fn(),
  get: jest.fn(),
  post: jest.fn(),
  put: jest.fn(),
  delete: jest.fn(),
  interceptors: {
    request: {
      use: jest.fn(),
      eject: jest.fn()
    },
    response: {
      use: jest.fn()
    }
  }
} as any;
