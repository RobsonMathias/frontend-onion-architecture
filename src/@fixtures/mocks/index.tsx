export * from './store.mock';
export * from './router.mock';
export * from './component.mock';
export * from './wrappers';
export * from './http.mock';
export * from './services.mock';
export * from './navigation.mock';
export * from './storage.mock';
export * from './use-auth.mock';
