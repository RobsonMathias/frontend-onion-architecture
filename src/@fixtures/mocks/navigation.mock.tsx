import React from 'react';

const setData = jest.fn();
const mockUseUI = jest.fn(() => ({ setData }));

const query = { get: jest.fn(), set: jest.fn() };
const set = jest.fn();
const setAll = jest.fn();
const mockUseQuery = jest.fn(() => ({ query, setAll, set }));

const replace = jest.fn();
const mockUseURL = jest.fn(() => ({ replace }));

const mockNavBar = React.forwardRef((props, ref) => <div {...props} />);

export const mockNavigation = {
  NavBar: mockNavBar,
  useQuery: mockUseQuery,
  query,
  setAll,
  set,
  useUI: mockUseUI,
  setData,
  useURL: mockUseURL,
  replace
};
