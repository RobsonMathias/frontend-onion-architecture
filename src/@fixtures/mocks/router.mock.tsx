import React from 'react';

function Mock({ children }: any) {
  return <>{children}</>;
}

const mockUseLocation = jest.fn(() => ({}));
const goBack = jest.fn();
const push = jest.fn();
const listen = jest.fn();
const mockUseHistory = jest.fn(() => ({ push, goBack, location: {}, listen, action: 'ACTION' }));
const useRouteMatch = () => ({ push: jest.fn(), url: '/mocked' });
const mockUseParams = jest.fn();
const mockNavLink = React.forwardRef(
  ({ to, children, activeClassName, className, 'data-testid': testId }: any, ref) => (
    <a
      className={`${className}-${activeClassName}`}
      href={to}
      children={children}
      data-testid={testId}
    />
  )
);
const Link = React.forwardRef((props, ref) => <p {...props} />);
const mockRoute = React.forwardRef((props, ref) => <div />);
const mockSwitch = React.forwardRef((props, ref) => <div />);
jest.mock('react-router-dom', () => ({
  Redirect: Mock,
  BrowserRouter: Mock,
  useHistory: mockUseHistory,
  useParams: mockUseParams,
  useRouteMatch,
  NavLink: mockNavLink,
  Route: mockRoute,
  Switch: mockSwitch,
  useLocation: mockUseLocation
}));

export const mockRouter = {
  useHistory: mockUseHistory,
  Link,
  useParams: mockUseParams,
  NavLink: mockNavLink,
  useLocation: mockUseLocation,
  goBack,
  push,
  listen
};
