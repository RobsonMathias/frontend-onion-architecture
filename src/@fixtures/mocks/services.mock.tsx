import { Service } from 'services/service';

const auth = {
  onAuthStateChange: jest.fn(),
  signOut: jest.fn(),
  signIn: jest.fn(),
  signInWithGoogle: jest.fn(),
  signUp: jest.fn()
};

const task = {
  list: jest.fn(),
  create: jest.fn(),
  update: jest.fn(),
  delete: jest.fn()
};

function ServicesProvider({ children }: any) {
  return <div>{children}</div>;
}

const instances = {
  auth,
  task
};
const mock = {
  ServicesProvider,
  useService: jest.fn(() => instances)
};
jest.doMock('services', () => mock);

export const mockServices = {
  mock,
  auth,
  task
};

export const mockServiceInstance = {
  patch: jest.fn(),
  get: jest.fn(),
  post: jest.fn(),
  put: jest.fn(),
  delete: jest.fn(),
  setInterceptorHeaders: jest.fn(),
  removeInterceptorHeaders: jest.fn()
} as any;
