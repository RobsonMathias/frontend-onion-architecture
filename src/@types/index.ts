declare global {
  interface Window {
    localStorage: any;
    sessionStorage: any;
  }
}

export * from './task';
export * from './store';
export * from './auth';
export * from './storage';
export * from './analytics';
