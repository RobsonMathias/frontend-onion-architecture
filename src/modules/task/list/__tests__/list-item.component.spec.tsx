/**
 * @jest-environment jsdom
 */
import '@testing-library/jest-dom/extend-expect';

import { act, fireEvent, render } from '@testing-library/react';
import '@fixtures/mocks';
import React from 'react';

import { ListItemComponent, ListItemComponentProps } from '../list-item.component';
import { TaskStatus } from '@types';

const onEvent = jest.fn();
const defaultProps: ListItemComponentProps = {
  task: {
    id: 'id',
    status: TaskStatus.NEW,
    description: 'description'
  },
  onEvent
};

describe('ListItemComponent', () => {
  it('should render successfully', () => {
    const { container } = render(<ListItemComponent {...defaultProps} />);
    expect(container).toMatchSnapshot();
  });

  it('should call onEvent with remove successfully', async () => {
    const { getByTestId } = render(<ListItemComponent {...defaultProps} />);

    await act(async () => {
      fireEvent.click(getByTestId('remove'));
    });

    expect(onEvent).toHaveBeenCalledWith(
      {
        id: 'id',
        status: 'new',
        description: 'description'
      },
      'remove'
    );
  });

  it('should call onEvent with edit successfully', async () => {
    const { getByTestId } = render(<ListItemComponent {...defaultProps} />);

    await act(async () => {
      fireEvent.click(getByTestId('edit'));
    });

    expect(onEvent).toHaveBeenCalledWith(
      {
        id: 'id',
        status: 'new',
        description: 'description'
      },
      'edit'
    );
  });

  it('should call onEvent with edit successfully', async () => {
    const { getByTestId } = render(<ListItemComponent {...defaultProps} />);

    await act(async () => {
      fireEvent.click(getByTestId('edit'));
    });

    expect(onEvent).toHaveBeenCalledWith(
      {
        id: 'id',
        status: 'new',
        description: 'description'
      },
      'edit'
    );
  });

  it('should call onEvent with status successfully', async () => {
    const { getByTestId } = render(<ListItemComponent {...defaultProps} />);

    await act(async () => {
      fireEvent.change(getByTestId('status'), { target: { value: 'done' } });
    });

    expect(onEvent).toHaveBeenCalledWith(
      {
        id: 'id',
        status: 'done',
        description: 'description'
      },
      'status'
    );
  });
});
