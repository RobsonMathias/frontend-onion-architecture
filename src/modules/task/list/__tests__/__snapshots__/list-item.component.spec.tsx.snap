// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItemComponent should render successfully 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container css-9cyib4-MuiGrid-root"
      >
        <h6
          class="MuiTypography-root MuiTypography-subtitle1 css-l57v1u-MuiTypography-root"
          data-testid="edit"
        >
          description
        </h6>
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-labelledby="task-status-label-id task-status-id"
            class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
            id="task-status-id"
            role="button"
            tabindex="0"
          >
            Novo
          </div>
          <input
            aria-hidden="true"
            class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
            data-testid="status"
            tabindex="-1"
            value="new"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-173wfxe"
            >
              <span>
                Status
              </span>
            </legend>
          </fieldset>
        </div>
        <button
          aria-label="Remover"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          data-testid="remove"
          tabindex="0"
          type="button"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M5 20a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V8h2V6h-4V4a2 2 0 0 0-2-2H9a2 2 0 0 0-2 2v2H3v2h2zM9 4h6v2H9zM8 8h9v12H7V8z"
            />
            <path
              d="M9 10h2v8H9zm4 0h2v8h-2z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
